### ISSUE TITLE
Recent versions doesn't like "src" folder in the root

### ISSUE THREAD

**Issue by xixiaofinland** [dangmai/prettier-plugin-apex#1706]
### Self-Troubleshooting Checklist

- [X] Are you running Prettier Apex >= 2.0.0? If Yes, have you gone through the [Upgrade Guide](https://github.com/dangmai/prettier-plugin-apex/wiki/Upgrading-to-Prettier-Apex-v2)?
- [X] Are you running VSCode and recently upgraded Prettier or Prettier Apex? If Yes, have you restarted VSCode after the upgrade?

### Describe your issue

I use the global package and run for example: `npx prettier --plugin=prettier-plugin-apex --parser=apex if1.cls`

It worked in a couple of versions ago but failed to execute recently.
After investigation, I find out that it doesn't like a "src" folder in the root, which I do have in my repo: https://github.com/xixiaofinland/afmt

After deleting "src" folder, it starts to work normally.

It gives error as below. 

```
[error] warning: `afmt` (bin "afmt") generated 2 warnings (run `cargo fix --bin "afmt"` to apply 2 suggestions)
[error]     Finished `dev` profile [optimized + debuginfo] target(s) in 1.82s
[error]      Running `target/debug/afmt`
[error] node_modules/prettier-plugin-apex/vendor/apex-ast-serializer/bin/apex-ast-serializer: line 251: public: command not found
[error] node_modules/prettier-plugin-apex/vendor/apex-ast-serializer/bin/apex-ast-serializer: eval: line 253: syntax error near unexpected token `Integer'
[error] node_modules/prettier-plugin-apex/vendor/apex-ast-serializer/bin/apex-ast-serializer: eval: line 253: `    if (true) Integer i = 1; else if (false) Integer l = 3; else Integer j = 2
```

### System Info

```shell
npx envinfo --system --npmPackages '?(@)prettier*/**' --binaries --npmGlobalPackages '?(@)prettier*/**' --fullTree --languages
Need to install the following packages:
envinfo@7.14.0
Ok to proceed? (y) n
```

**Comment by dangmai** [comment]
Do you mind allowing the `envinfo` command to run? Without it, it's a bit difficult for me to understand your machine's configuration and the versions involved

**Comment by dangmai** [comment]
I tried to reproduce this in this repo: https://github.com/dangmai/test-ppa-1706 but couldn't. That repo has Apex files in both `src` and `tests` dir, and both can be formatted without issue.

**Comment by xixiaofinland** [comment]
Here is more info below. It's a really interesting issue to me. 

When I execute `npx prettier`, it first prints the debug that `Rust run/compile` would do, then parses to AST and obviously fails. 

I have little idea why it will invoke rust despite they site in the same repo. 

```
~/projects/afmt poc*
Afmt-env Γ¥» npx prettier --plugin=prettier-plugin-apex --parser=apex if1.cls --config tests/configs/.prettierrc_p80.toml
if1.cls
[error] if1.cls: Error: warning: unused import: `Arc`
[error]  --> src/formatter.rs:8:23
[error]   |
[error] 8 | use std::sync::{mpsc, Arc};
[error]   |                       ^^^
[error]   |
[error]   = note: `#[warn(unused_imports)]` on by default
[error]
[error] warning: unused variable: `indent`
[error]    --> src/data_model.rs:432:28
[error]     |
[error] 432 |     pub fn new(node: Node, indent: usize) -> Self {
[error]     |                            ^^^^^^ help: if this is intentional, prefix it with an underscore: `_indent`
[error]     |
[error]     = note: `#[warn(unused_variables)]` on by default
[error]
[error] warning: unused variable: `indent`
[error]    --> src/data_model.rs:446:28
[error]     |
[error] 446 |     pub fn new(node: Node, indent: usize) -> Self {
[error]     |                            ^^^^^^ help: if this is intentional, prefix it with an underscore: `_indent`
[error]
[error] warning: associated function `new` is never used
[error]    --> src/data_model.rs:432:12
[error]     |
[error] 431 | impl ArrayInitializer {
[error]     | --------------------- associated function in this implementation
[error] 432 |     pub fn new(node: Node, indent: usize) -> Self {
[error]     |            ^^^
[error]     |
[error]     = note: `#[warn(dead_code)]` on by default
[error]
[error] warning: associated function `new` is never used
[error]    --> src/data_model.rs:446:12
[error]     |
[error] 445 | impl AssignmentExpression {
[error]     | ------------------------- associated function in this implementation
[error] 446 |     pub fn new(node: Node, indent: usize) -> Self {
[error]     |            ^^^
[error]
[error] warning: multiple variants are never constructed
[error]    --> src/enum_def.rs:270:5
[error]     |
[error] 269 | pub enum Modifier {
[error]     |          -------- variants in this enum
[error] 270 |     Abstract,
[error]     |     ^^^^^^^^
[error] 271 |     Final,
[error]     |     ^^^^^
[error] 272 |     Global,
[error]     |     ^^^^^^
[error] 273 |     InheritedSharing,
[error]     |     ^^^^^^^^^^^^^^^^
[error] ...
[error] 276 |     Protected,
[error]     |     ^^^^^^^^^
[error] ...
[error] 279 |     TestMethod,
[error]     |     ^^^^^^^^^^
[error] 280 |     Transient,
[error]     |     ^^^^^^^^^
[error] 281 |     Virtual,
[error]     |     ^^^^^^^
[error] 282 |     Webservice,
[error]     |     ^^^^^^^^^^
[error]     |
[error]     = note: `Modifier` has a derived impl for the trait `Debug`, but this is intentionally ignored during dead code analysis
[error]
[error] warning: `afmt` (lib) generated 6 warnings (run `cargo fix --lib -p afmt` to apply 1 suggestion)
[error] warning: unused import: `anyhow`
[error]  --> src/main.rs:4:14
[error]   |
[error] 4 | use anyhow::{anyhow, Result};
[error]   |              ^^^^^^
[error]   |
[error]   = note: `#[warn(unused_imports)]` on by default
[error]
[error] warning: unused import: `fs`
[error]  --> src/main.rs:9:11
[error]   |
[error] 9 | use std::{fs, process};
[error]   |           ^^
[error]
[error] warning: `afmt` (bin "afmt") generated 2 warnings (run `cargo fix --bin "afmt"` to apply 2 suggestions)
[error]     Finished `dev` profile [optimized + debuginfo] target(s) in 0.04s
[error]      Running `target/debug/afmt`
[error] node_modules/prettier-plugin-apex/vendor/apex-ast-serializer/bin/apex-ast-serializer: line 251: public: command not found
[error] node_modules/prettier-plugin-apex/vendor/apex-ast-serializer/bin/apex-ast-serializer: eval: line 253: syntax error near unexpected token `Integer'
[error] node_modules/prettier-plugin-apex/vendor/apex-ast-serializer/bin/apex-ast-serializer: eval: line 253: `    if (true) Integer i = 1; else if (false) Integer l = 3; else Integer j = 2;'
[error] Usage: java [options] <mainclass> [args...]
[error]            (to execute a class)
[error]    or  java [options] -jar <jarfile> [args...]
[error]            (to execute a jar file)
[error]    or  java [options] -m <module>[/<mainclass>] [args...]
[error]        java [options] --module <module>[/<mainclass>] [args...]
[error]            (to execute the main class in a module)
[error]    or  java [options] <sourcefile> [args]
[error]            (to execute a single source-file program)
[error]
[error]  Arguments following the main class, source file, -jar <jarfile>,
[error]  -m or --module <module>/<mainclass> are passed as the arguments to
[error]  main class.
[error]
[error]  where options include:
[error]
[error]     -cp <class search path of directories and zip/jar files>
[error]     -classpath <class search path of directories and zip/jar files>
[error]     --class-path <class search path of directories and zip/jar files>
[error]                   A : separated list of directories, JAR archives,
[error]                   and ZIP archives to search for class files.
[error]     -p <module path>
[error]     --module-path <module path>...
[error]                   A : separated list of elements, each element is a file path
[error]                   to a module or a directory containing modules. Each module is either
[error]                   a modular JAR or an exploded-module directory.
[error]     --upgrade-module-path <module path>...
[error]                   A : separated list of elements, each element is a file path
[error]                   to a module or a directory containing modules to replace
[error]                   upgradeable modules in the runtime image. Each module is either
[error]                   a modular JAR or an exploded-module directory.
[error]     --add-modules <module name>[,<module name>...]
[error]                   root modules to resolve in addition to the initial module.
[error]                   <module name> can also be ALL-DEFAULT, ALL-SYSTEM,
[error]                   ALL-MODULE-PATH.
[error]     --enable-native-access <module name>[,<module name>...]
[error]                   modules that are permitted to perform restricted native operations.
[error]                   <module name> can also be ALL-UNNAMED.
[error]     --list-modules
[error]                   list observable modules and exit
[error]     -d <module name>
[error]     --describe-module <module name>
[error]                   describe a module and exit
[error]     --dry-run     create VM and load main class but do not execute main method.
[error]                   The --dry-run option may be useful for validating the
[error]                   command-line options such as the module system configuration.
[error]     --validate-modules
[error]                   validate all modules and exit
[error]                   The --validate-modules option may be useful for finding
[error]                   conflicts and other errors with modules on the module path.
[error]     -D<name>=<value>
[error]                   set a system property
[error]     -verbose:[class|module|gc|jni]
[error]                   enable verbose output for the given subsystem
[error]     -version      print product version to the error stream and exit
[error]     --version     print product version to the output stream and exit
[error]     -showversion  print product version to the error stream and continue
[error]     --show-version
[error]                   print product version to the output stream and continue
[error]     --show-module-resolution
[error]                   show module resolution output during startup
[error]     -? -h -help
[error]                   print this help message to the error stream
[error]     --help        print this help message to the output stream
[error]     -X            print help on extra options to the error stream
[error]     --help-extra  print help on extra options to the output stream
[error]     -ea[:<packagename>...|:<classname>]
[error]     -enableassertions[:<packagename>...|:<classname>]
[error]                   enable assertions with specified granularity
[error]     -da[:<packagename>...|:<classname>]
[error]     -disableassertions[:<packagename>...|:<classname>]
[error]                   disable assertions with specified granularity
[error]     -esa | -enablesystemassertions
[error]                   enable system assertions
[error]     -dsa | -disablesystemassertions
[error]                   disable system assertions
[error]     -agentlib:<libname>[=<options>]
[error]                   load native agent library <libname>, e.g. -agentlib:jdwp
[error]                   see also -agentlib:jdwp=help
[error]     -agentpath:<pathname>[=<options>]
[error]                   load native agent library by full pathname
[error]     -javaagent:<jarpath>[=<options>]
[error]                   load Java programming language agent, see java.lang.instrument
[error]     -splash:<imagepath>
[error]                   show splash screen with specified image
[error]                   HiDPI scaled images are automatically supported and used
[error]                   if available. The unscaled image filename, e.g. image.ext,
[error]                   should always be passed as the argument to the -splash option.
[error]                   The most appropriate scaled image provided will be picked up
[error]                   automatically.
[error]                   See the SplashScreen API documentation for more information
[error]     @argument files
[error]                   one or more argument files containing options
[error]     --disable-@files
[error]                   prevent further argument file expansion
[error]     --enable-preview
[error]                   allow classes to depend on preview features of this release
[error] To specify an argument for a long option, you can use --<name>=<value> or
[error] --<name> <value>.
[error]
[error]
[error]     at ChildProcess.<anonymous> (file:///home/nixos/projects/afmt/node_modules/prettier-plugin-apex/dist/src/parser.js:38:24)
[error]     at ChildProcess.emit (node:events:520:28)
[error]     at maybeClose (node:internal/child_process:1105:16)
[error]     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
```

**Comment by xixiaofinland** [comment]
When I rename `src/` to `src1/`, it starts to work.

```
~/projects/afmt poc*
Afmt-env Γ¥» la
Permissions Size User  Date Modified Name
.rw-r--r--   121 nixos 11 Nov 22:11  .afmt.toml_back
drwxr-xr-x     - nixos 11 Nov 22:11  .direnv
.rw-r--r--    31 nixos 11 Nov 22:11  .envrc
drwxr-xr-x     - nixos 11 Nov 23:38  .git
.rw-r--r--   265 nixos 11 Nov 22:11  .gitattributes
drwxr-xr-x     - nixos 11 Nov 22:11  .github
.rw-r--r--   122 nixos 11 Nov 22:20  .gitignore
drwxr-xr-x     - nixos 11 Nov 22:20  benches
.rw-r--r--   658 nixos 11 Nov 22:11  build.rs
.rw-r--r--   26k nixos 11 Nov 22:20  Cargo.lock
.rw-r--r--   603 nixos 11 Nov 22:20  Cargo.toml
drwxr-xr-x     - nixos 11 Nov 22:11  dep
drwxr-xr-x     - nixos 11 Nov 22:11  docs
.rw-r--r--   134 nixos 11 Nov 22:41  if1.cls
drwxr-xr-x     - nixos 11 Nov 22:11  md
drwxr-xr-x     - nixos 11 Nov 22:41  node_modules
.rw-r--r--   13k nixos 11 Nov 22:43  package-lock.json
.rw-r--r--   345 nixos 11 Nov 22:41  package.json
.rw-r--r--  2.8k nixos 11 Nov 22:11  README.md
.rw-r--r--    29 nixos 11 Nov 22:11  rustfmt.toml
drwxr-xr-x     - nixos 11 Nov 22:11  samples
drwxr-xr-x     - nixos 11 Nov 22:44  src
drwxr-xr-x     - nixos 11 Nov 22:11  target
.rw-r--r--   135 nixos 11 Nov 22:20  test.cls
drwxr-xr-x     - nixos 11 Nov 22:20  tests

~/projects/afmt poc*
Afmt-env Γ¥» mv src src1

~/projects/afmt poc*
Afmt-env Γ¥» npx prettier --plugin=prettier-plugin-apex --parser=apex if1.cls --config tests/configs/.prettierrc_p80.toml
public class Me {
  {
    if (true)
      Integer i = 1;
    else if (false)
      Integer l = 3;
    else
      Integer j = 2;
  }
}

~/projects/afmt poc*
Afmt-env Γ¥» npx envinfo --system --npmPackages '?(@)prettier*/**' --binaries --npmGlobalPackages '?(@)prettier*/**' --fullTree --languages
Need to install the following packages:
envinfo@7.14.0
Ok to proceed? (y) y


  System:
    OS: Linux 5.15 NixOS 24.11 (Vicuna) 24.11 (Vicuna)
    CPU: (4) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 7.16 GB / 7.76 GB
    Container: Yes
    Shell: 5.9 - /run/current-system/sw/bin/zsh
  Binaries:
    Node: 22.8.0 - /nix/store/xsrv93rg8vi05k9ig6v9mcvcyn01n6n2-nodejs-22.8.0/bin/node
    npm: 10.8.2 - /nix/store/xsrv93rg8vi05k9ig6v9mcvcyn01n6n2-nodejs-22.8.0/bin/npm
  Languages:
    Bash: 5.2.32 - /nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/bash
    Java: 21.0.3 - /nix/store/59flqcj6x3dxiwjavxkwrycamg0482yb-openjdk-21.0.3+9/bin/javac
    Perl: 5.38.2 - /run/current-system/sw/bin/perl
    Rust: 1.81.0 - /nix/store/mz68k40jyzrq6r5pjfsk0jcp0a9yniyl-rust-default-1.81.0/bin/rustc
  npmPackages:
    @prettier-apex/apex-ast-serializer-linux-x64:  2.2.1
    prettier: ^3.3.3 => 3.3.3
    prettier-plugin-apex: ^2.2.1 => 2.2.1
```****

**Comment by xixiaofinland** [comment]
if I don't set `apexStandaloneParser` to `none` to use the old way, NixOS won't let it run.


```
Afmt-env Γ¥» npx prettier --plugin=prettier-plugin-apex --parser=apex if1.in
if1.in
[error] if1.in: Error: Could not start dynamically linked executable: node_modules/@prettier-apex/apex-ast-serializer-linux-x64/apex-ast-serializer-linux-x64
[error] NixOS cannot run dynamically linked executables intended for generic
[error] linux environments out of the box. For more information, see:
[error] https://nix.dev/permalink/stub-ld
[error]
[error]     at ChildProcess.<anonymous> (file:///home/nixos/projects/afmt/node_modules/prettier-plugin-apex/dist/src/parser.js:38:24)
[error]     at ChildProcess.emit (node:events:520:28)
[error]     at maybeClose (node:internal/child_process:1105:16)
[error]     at Socket.<anonymous> (node:internal/child_process:457:11)
[error]     at Socket.emit (node:events:520:28)
[error]     at Pipe.<anonymous> (node:net:338:12)
```

**Comment by dangmai** [comment]
That's actually very interesting - I didn't realize NixOS has that restriction. Let me see how much bigger the binaries are going to be if I static-link everything.

**Comment by xixiaofinland** [comment]
@dangmai  sorry, I removed theout-dated info, but great you have seen them.

Now I spotted out that in my repo, npx prettier will secretly invoke [build.rs](https://github.com/xixiaofinland/afmt/blob/poc/build.rs) which subsequently calls [parser.c](https://github.com/xixiaofinland/afmt/blob/poc/dep/parser.c) in the code, I don't know why it will do so.

`npx prettier --plugin=prettier-plugin-apex --parser=apex if1.in --config tests/configs/.prettierrc_p80.toml`

**Comment by dangmai** [comment]
If you run:

```
npx prettier --plugin=prettier-plugin-apex --parser=apex --apex-standalone-parser=none if1.in
```

Does it invoke Rust? I am so curious what could've caused that

**Comment by xixiaofinland** [comment]
> If you run:
> 
> ```
> npx prettier --plugin=prettier-plugin-apex --parser=apex --apex-standalone-parser=none if1.in
> ```
> 
> Does it invoke Rust? I am so curious what could've caused that

same error

**Comment by xixiaofinland** [comment]
Ah, it is indeed from `.direnv` folder made by `.envrc`!

here is the[ .envrc file](https://github.com/xixiaofinland/afmt/blob/main/.envrc)
and it loads [this config](https://github.com/xixiaofinland/dotfiles-nix/blob/f36a3bc86899d67795d73634632eaf8d1c75e842/flake.nix#L213)

Most likely it's one of those alias shortcuts I made.... so close to the root cause now :D

**Comment by xixiaofinland** [comment]
I removed all those alias, and it starts to work normally. Just need time to identify which one is the criminal now :D.
Thanks a lot to point it out!

**Comment by dangmai** [comment]
Nice, I'm glad you have some avenue to debug this further. I'll try out the static-link binaries approach and will let you know when one's available, so you can take advantage of the faster parser

**Comment by dangmai** [comment]
Could you try this file: https://github.com/dangmai/prettier-plugin-apex/actions/runs/11788250021/artifacts/2173820974 to see if it works in the NixOS environment? You can extract it (and probably do `chmod +x` on the extracted file) and copy into `node_modules/@prettier-apex/apex-ast-serializer-linux-x64/`.

If it works I'll cut a new release 😄

**Comment by xixiaofinland** [comment]
it works! And I can feel the returning speed difference between the two cmds. @dangmai 

npx prettier --plugin=prettier-plugin-apex --parser=apex --apex-standalone-parser=none if1.cls
npx prettier --plugin=prettier-plugin-apex --parser=apex if1.cls

**Comment by dangmai** [comment]
I've released `v2.2.2` that contains static-linked executable. Let me know if that works for you, good luck! 😃

**Comment by xixiaofinland** [comment]
I confirm v2.2.2 works like a charm for me on NixOS, good work!
